### ISSUE TITLE
feat(timeseries): add TrainWindowed/PredictWindowed to PatchTST

### ISSUE THREAD

**Issue by dndungu** [zerfoo/zerfoo#120]
## Context

DLinear, N-HiTS, and CfC all implement TrainWindowed/PredictWindowed for CPU-based training on temporal windows. PatchTST only has a Forward/Predict method — it's missing the standalone training loop.

## Current State

- `DLinear.TrainWindowed` — implemented
- `NHiTS.TrainWindowed` — implemented
- `CfC.TrainWindowed` — implemented
- `PatchTST.TrainWindowed` — **missing**

## Requested API

```go
func (m *PatchTST) TrainWindowed(windows [][][]float64, labels []float64, config TrainConfig) (*TrainResult, error)
func (m *PatchTST) PredictWindowed(modelPath string, windows [][][]float64) ([]float64, error)
```

Should use analytical backpropagation through the transformer attention layers (not finite-difference). Include AdamW optimizer, gradient clipping, and z-score normalization.
